File: CarRentalClient/src/app/shared/models/Order.model.ts
import { CarModel } from "./Car.Model";
import { UserModel } from "./User.model";
import { formatDate } from "@angular/common";

export interface OrderModel {

    OrderID: number;
    OrderDate: Date;
    Car: CarModel;
    UserID: UserModel;
    OrderStart: Date;
    PlannedEnd: Date;
    ActualEnd: Date;
    OrigianlCost: number;
    ActualCost: number;
    IsActiveOrder: boolean;
}

File: CarRentalServer/02_BOL/OrderModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace _02_BOL
{
    public class OrderModel
    {

        public int OrderID { get; set; }

        [Required]
        public System.DateTime OrderDate { get; set; }

        [Required]
        public CarModel Car { get; set; }

        [Required]
        public UserModel UserID { get; set; }

        [Required]
        public System.DateTime OrderStart { get; set; }

        [Required]
        public System.DateTime PlannedEnd { get; set; }


        public System.DateTime? ActualEnd { get; set; } //because in Db can be null

        [Required]
        public decimal OrigianlCost { get; set; }


        public decimal? ActualCost { get; set; }//because in Db can be null
        public bool IsActiveOrder { get; set; }




    }
}
